azext_iot/iothub/edge_device_config.py

```diff
@@ -0,0 +1,469 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
```

Comment:
Seems appropriate to put this module in providers/, thoughts?

Reply:
Now that I've finished a bunch of other renaming / changes - I'm still thinking about this.

Right now the providers folder only contains actual provider modules (`base`, `device_identity`, `device_messaging`, etc.).

Since this code is used almost exclusively by the new `device_identity` provider, but isn't a provider itself, it seems odd to just throw it in that folder... would it make sense to move some providers into folders (`providers/device_identity/device_identity_provider.py`) in case we extend them? Or maybe just another folder inside `providers` - `providers/[utility | common | helpers]`?

Reply:
The first option (`providers/device_identity/device_identity_provider.py`) is not my favorite because it looks too nested and repeated.

`from iothub.providers.device_identity.device_identity_provider` ...I know I'm importing an iothub provider from the first 2 segments.

And IMO we don't have enough independent modules around device identity at this stage to create a new nested directory.

Then `common` is overloaded - what about for now, putting it in a new helpers directory as a sibling to providers ? i.e.  ....`iothub/helpers/edge_device_config.py`?